src/api/Feed.js

```diff
@@ -394,38 +478,77 @@ class Feed extends Base {
+        const appActivityPromise =
+            !shouldUseUAA && shouldShowAppActivity ? this.fetchAppActivity(permissions) : Promise.resolve();
+
+        const fileActivitiesPromise = shouldUseUAA
+            ? this.fetchFileActivities(permissions, [
+                  FILE_ACTIVITY_TYPE_ANNOTATION,
+                  FILE_ACTIVITY_TYPE_APP_ACTIVITY,
+                  FILE_ACTIVITY_TYPE_COMMENT,
+                  FILE_ACTIVITY_TYPE_TASK,
+              ])
             : Promise.resolve();
-        const versionsPromise = shouldShowVersions ? this.fetchVersions() : Promise.resolve();
-        const currentVersionPromise = shouldShowVersions ? this.fetchCurrentVersion() : Promise.resolve();
-        const commentsPromise = shouldShowReplies
-            ? this.fetchThreadedComments(permissions)
-            : this.fetchComments(permissions);
-        const tasksPromise = shouldShowTasks ? this.fetchTasksNew() : Promise.resolve();
-        const appActivityPromise = shouldShowAppActivity ? this.fetchAppActivity(permissions) : Promise.resolve();
-
-        Promise.all([
-            versionsPromise,
-            currentVersionPromise,
-            commentsPromise,
-            tasksPromise,
-            appActivityPromise,
-            annotationsPromise,
-        ]).then(([versions: ?FileVersions, currentVersion: ?BoxItemVersion, ...feedItems]) => {
-            const versionsWithCurrent = currentVersion
-                ? this.versionsAPI.addCurrentVersion(currentVersion, versions, this.file)
-                : undefined;
-            const sortedFeedItems = sortFeedItems(versionsWithCurrent, ...feedItems);
+
+        const handleFeedItems = (feedItems: FeedItems) => {
             if (!this.isDestroyed()) {
-                this.setCachedItems(id, sortedFeedItems);
+                this.setCachedItems(id, feedItems);
                 if (this.errors.length) {
-                    errorCallback(sortedFeedItems, this.errors);
+                    errorCallback(feedItems, this.errors);
                 } else {
-                    successCallback(sortedFeedItems);
+                    successCallback(feedItems);
                 }
             }
-        });
+        };
+
+        if (shouldUseUAA) {
+            Promise.all([versionsPromise, currentVersionPromise, fileActivitiesPromise]).then(
+                ([versions: ?FileVersions, currentVersion: ?BoxItemVersion, ...feedItems]) => {
+                    if (!feedItems || !feedItems.length) {
+                        return;
+                    }
+
+                    const fileActivitiesResponse = feedItems[0];
+                    const versionsWithCurrent = currentVersion
+                        ? this.versionsAPI.addCurrentVersion(currentVersion, versions, this.file)
+                        : undefined;
+                    const parsedFeedItems = parseFileActivitiesResponseForFeed(fileActivitiesResponse);
+                    // $FlowFixMe
```

Comment:
Why is this needed?

Reply:
Flow is complaining that Comments/Annotations/Tasks, etc. is incompatable with an empty array literal. I've tried allowing an empty array as a parameter in `parseFileActivitiesResponseForFeed` however that did not fix the problem either...

`Cannot call sortFeedItems because:
 • in array element:
    • Either empty array literal [1] is incompatible with Comments [2].
    • Or empty array literal [1] is incompatible with ThreadedComments [3].
    • Or empty array literal [1] is incompatible with Tasks [4].
    • Or empty array literal [1] is incompatible with FileVersions [5].
    • Or empty array literal [1] is incompatible with AppActivityItems [6].
    • Or empty array literal [1] is incompatible with Annotations [7].
 • in array element:
    • Either property total_count is missing in object literal [8] but exists in Comments [2].
    • Or property limit is missing in object literal [8] but exists in ThreadedComments [3].
    • Or property next_marker is missing in object literal [8] but exists in Tasks [4].
    • Or property total_count is missing in object literal [8] but exists in FileVersions [5].
    • Or property total_count is missing in object literal [8] but exists in AppActivityItems [6].
    • Or property limit is missing in object literal [8] but exists in Annotations [7].`